platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+---
+layout: page
+title: Platform-specific APIs and Platform Plugins
+permalink: /platform-plugins/
+---
+
+This guide describes how to connect your Flutter app to platform-specific
+services available on iOS and Android devices. This includes both device APIs
+(like `Clipboard` and `BatteryState`) and third-party platform SDKs (like
+`Google Analytics`).
+
+* TOC
+{:toc}
+
+## Options for calling platform-specific APIs
+
+Mobile devices offer many options to extend your app through platform-specific
+APIs. These are critical to the completeness of most mobile apps, and Flutter
+imposes no limits on your ability to them.
+
+Flutter offers the following ways to use platform-specific services:
+
+* **Use existing Platform Plugins:** For frequently used APIs and SDKs, use
+ existing Platform Plugins, with no need to write any platform-specific code.
+ [See details below](#use).
+
+* **Create a Platform Plugin:** Write your own platform-specific code using
+ Flutter's Platform Channels. This can be used for private APIs, or for cases
+ where a suitable Platform Plugin is not yet available. [See details below](#create).
+
+* **Publish a Platform Plugin:** Publish your own custom platform-specific as a
+ plugin. This enables others to use it by simply declaring a dependency on the
+ plugin. [See details below](#publish).
+
+## Use existing Platform Plugins {#use}
+
+### Core Flutter Platform Plugin
```

Comment:
So, yes this is confusing mostly as we have stuff in services that really should just be in an optional plugin.